src/rpc/blockchain.cpp

```diff
@@ -1570,6 +1571,286 @@ UniValue getchaintxstats(const JSONRPCRequest& request)
+            "  \"txs\": xxxxx,             (numeric) The number of transactions (excluding coinbase).\n"
+            "  \"swtxs\": xxxxx,           (numeric) The number of segwit transactions.\n"
+            "  \"ins\": xxxxx,             (numeric) The number of inputs (excluding coinbase).\n"
+            "  \"outs\": xxxxx,            (numeric) The number of outputs (including coinbase).\n"
+            "  \"subsidy\": xxxxx,         (numeric) The block subsidy.\n"
+            "  \"totalfee\": xxxxx,        (numeric) The fee total.\n"
+            "  \"utxo_increase\": xxxxx,   (numeric) The increase/decrease in the number of unspent outputs.\n"
+            "  \"utxo_size_inc\": xxxxx,   (numeric) The increase/decrease in size for the utxo index (not discounting op_return and similar).\n"
+            "  \"total_size\": xxxxx,      (numeric) Total size of all non-coinbase transactions.\n"
+            "  \"total_weight\": xxxxx,    (numeric) Total weight of all non-coinbase transactions divided by segwit scale factor (4).\n"
+            "  \"swtotal_size\": xxxxx,    (numeric) Total size of all segwit transactions.\n"
+            "  \"swtotal_weight\": xxxxx,  (numeric) Total weight of all segwit transactions divided by segwit scale factor (4).\n"
+            "  \"total_out\": xxxxx,       (numeric) Total amount in all outputs (excluding coinbase and thus reward [ie subsidy + totalfee]).\n"
+            "  \"minfee\": xxxxx,          (numeric) Minimum fee in the block.\n"
+            "  \"maxfee\": xxxxx,          (numeric) Maximum fee in the block.\n"
+            "  \"medianfee\": xxxxx,       (numeric) Truncated median fee in the block.\n"
+            "  \"avgfee\": xxxxx,          (numeric) Average fee in the block.\n"
+            "  \"minfeerate\": xxxxx,      (numeric) Minimum feerate (in satoshis per virtual byte).\n"
+            "  \"maxfeerate\": xxxxx,      (numeric) Maximum feerate (in satoshis per virtual byte).\n"
+            "  \"medianfeerate\": xxxxx,   (numeric) Truncated median feerate (in satoshis per virtual byte).\n"
+            "  \"avgfeerate\": xxxxx,      (numeric) Average feerate (in satoshis per virtual byte).\n"
+            "  \"mintxsize\": xxxxx,       (numeric) Minimum transaction size.\n"
+            "  \"maxtxsize\": xxxxx,       (numeric) Maximum transaction size.\n"
+            "  \"mediantxsize\": xxxxx,    (numeric) Truncated median transaction size.\n"
+            "  \"avgtxsize\": xxxxx,       (numeric) Average transaction size.\n"
+            "}\n"
+            "\nExamples:\n"
+            + HelpExampleCli("getblockstats", "1000 \"minfeerate,avgfeerate\"")
+            + HelpExampleRpc("getblockstats", "1000 \"maxfeerate,avgfeerate\"")
+        );
+
+    LOCK(cs_main);
+
+    int height = request.params[0].get_int();
+    int current_tip = chainActive.Height();
+    if (height < 0) {
+        height = current_tip + height;
```

Comment:
In commit "RPC: Introduce getblockstats":

Bug: This should be `current_tip + height + 1` so height -1 will return tip.